packages/react/src/components/Badge/Badge.tsx

```diff
@@ -1,40 +1,42 @@
 import cl from 'clsx/lite';
 import { type HTMLAttributes, type ReactNode, forwardRef } from 'react';
+import type { Color } from '../../colors';
 import type { DefaultProps } from '../../types';
+import type { Merge } from '../../utilities';
 
-export type BadgeProps = {
-  /**
-   * The color of the badge
-   *
-   * @default accent
-   */
-  color?: 'accent' | 'info' | 'success' | 'warning' | 'danger' | 'neutral';
-  /**
-   * The number to display in the badge
-   */
-  count?: number;
-  /**
-   * The maximum number to display in the badge, when the count exceeds this number, the badge will display `{max}+`
-   */
-  maxCount?: number;
-  /**
-   * The placement of the badge
-   *
-   * @default top-right
-   */
-  placement?: 'top-right' | 'top-left' | 'bottom-right' | 'bottom-left';
-  /**
-   * Use when badge is floating to change the position of the badge
-   *
-   * @default rectangle
-   */
-  overlap?: 'circle' | 'rectangle';
-  /**
-   * The badge will float on top of the children
-   */
-  children?: ReactNode;
-} & HTMLAttributes<HTMLSpanElement> &
-  DefaultProps;
+export type BadgeProps = Merge<
```

Comment:
Just curious/new to me 😇  why are we using `Merge`?

Reply:
It helps avoid a couple of issues:
  - Makes it impossible to forget to omit a overridden prop from `DefaultProps` or `HTMLAttributes`. Forgetting can lead to weird intersection types like `defaultValue?: string | (string & readonly string[])` in `ToggleGroup`, or plain non-working types like `data-size` when a component has defined it as something other than `"sm" | "md" | "lg"` because e.g. `("sm" | "md" | "lg") & ("sm" | "md" | "lg" | "xl")` === `"sm" | "md" | "lg"`
  - “Flattens” the type out so you get the props listed directly instead of a bunch of intersected types. This benefit is a bit more debatable, I guess, but I like it.

Reply:
nice! 🤩  so smart ❤️